FIX IterativeImputer skip iterative part if keep_empty_features is set to True (#29414)
scikit-learn/scikit-learn: doc/whats_new/v1.6.rst

  computing the mean value for uniform weights.
  :pr:`29135` by :user:`Xuefeng Xu <xuefeng-xu>`.

- |Fix| :class:`impute.IterativeImputer` Now gives same output 

>   when keep_empty_features is set to True, This is the same as when
>   keep_empty_features is to False.

Reviewer: ```suggestion
  when ``keep_empty_features=True`` compared to when
  ``keep_empty_features=False`` on non-empty features.
```
